src/Http/Cookie/Cookie.php

```diff
@@ -236,10 +236,22 @@ public function getValue()
     }
 
     /**
-     * {@inheritDoc}
-     * @psalm-suppress InvalidReturnType
+     * Gets the cookie value as a string.
+     *
+     * This will collapse any complex data in the cookie with json_encode()
+     *
+     * @return mixed
+     * @deprecated 4.0.0 Use getScalarValue() instead.
```

Comment:
When is the return not a string?

Reply:
Whenever it's not an array :)

Assertions in `ResponseTest::testWithCookieScalar()` (which I haven't changed) fail if the return value for the method is type hinted as `string`.